### PR TITLE
clientv3: send addresses forever until closed in simple balancer

### PR DESCRIPTION
**Pull request by mitake**
The channel returned by simpleBalancer.Notify() can be read multiple
times. So simpleBalancer should send addresses to the channel forever.

Fixes https://github.com/coreos/etcd/issues/5604

/cc @heyitsanthony

### CODE REVIEW

File: clientv3/balancer.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type simpleBalancer struct {
 	eps     []string
 	ch      chan []grpc.Address
 	numGets uint32
+	closeCh chan struct{}
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: most of the client code uses `donec` for this pattern

File: clientv3/balancer.go [hunk 2]
```diff
@@ -38,8 +39,22 @@ func newSimpleBalancer(eps []string) grpc.Balancer {
 	for i := range eps {
 		addrs[i].Addr = getHost(eps[i])
 	}
-	ch <- addrs
-	return &simpleBalancer{eps: eps, ch: ch}
+
+	closeCh := make(chan struct{})
+	go func() {
+		for {
+			select {
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: select {
case <-donec: return
case ch <- addrs:
}

File: clientv3/balancer.go [hunk 3]
```diff
@@ -38,8 +39,22 @@ func newSimpleBalancer(eps []string) grpc.Balancer {
...
+			case <-closeCh:
+				break
+			default:
+				ch <- addrs
+			}
+		}
+
+		close(closeCh)
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: not needed

File: clientv3/balancer.go [hunk 4]
```diff
@@ -38,8 +39,22 @@ func newSimpleBalancer(eps []string) grpc.Balancer {
...
+		close(ch)
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: `defer close(ch)` at the beginning of the goroutine

File: clientv3/balancer.go [hunk 5]
```diff
@@ -51,7 +66,7 @@ func (b *simpleBalancer) Get(ctx context.Context, opts grpc.BalancerGetOptions)
 }
 func (b *simpleBalancer) Notify() <-chan []grpc.Address { return b.ch }
 func (b *simpleBalancer) Close() error {
-	close(b.ch)
+	b.closeCh <- struct{}{}
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: close(b.donec)

**Comment by mitake** [comment]
@heyitsanthony updated for your comments, PTAL

**Comment by mitake** [comment]
Because of `ClientConn.lbWatcher()`, it is not a suitable solution. I'll open another PR later.

**Pull request closed without merging**